src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -352,10 +350,13 @@ private HttpResponse sendAndDefer(HttpRequest request, HttpRequestOptions option
       Response response = asyncHandler.getResponse();
       HttpResponse httpResponse = httpResponseCreator.create(response, inPipe);
       if (shouldFollowRedirect(httpResponse, options, enableMuleRedirect)) {
-        if (currentRedirects > MAX_REDIRECTS) {
+        if (currentRedirects >= MAX_REDIRECTS) {
           throw new IOException("Max redirects exceeded", new MaxRedirectException());
         }
-        httpResponse = sendAndDefer(createRedirectRequest(httpResponse, request), options, currentRedirects + 1);
+        HttpRequest redirectRequest = createRedirectRequest(httpResponse, request, options);
```

Comment:
Concern. Is this reuse of the same connection taken into account when this is the same host and protocol?

https://github.com/mulesoft/grizzly-ahc/blob/support/1_14-mule/src/main/java/com/ning/http/client/providers/grizzly/AhcEventFilter.java#L753

Not sure if this will derive in some functional problems. Some implementations of NTLM verify that you are using the same connection (though this will be part of the redirect). Anyway we should use the same connection whenever possible. Is there a way to reuse the connection if possible if we implement this in the http-service side.